### PR TITLE
Prepare for 0.4.1 release

### PR DESCRIPTION
**Pull request by NiallRees**
There have been a significant number of bug fixes already so I thought it was worth pushing a version.

### CODE REVIEW

**Review by pwildenhain — APPROVED** [review]
💯 Agree that a minor release is warranted

**Comment by alanmcruickshank** [comment]
As per #765 I think so long as we first merge #822, we should upgrade the project to "beta" on pypi from alpha. I've also enabled the homepage link too.

**Comment by NiallRees** [comment]
I've made a few tweaks to the README.md, with a headline update and change to the releases section so that it doesn't have to maintained going forward and just directs to the releases page. Feel free to suggest changes and tweaks https://github.com/sqlfluff/sqlfluff/blob/4913d72fd8a111f0402b449726aa6226e736a46e/README.md

File: setup.py [hunk 1]
```diff
@@ -57,7 +56,7 @@ def read(*names, **kwargs):
     include_package_data=True,
     classifiers=[
         # complete classifier list: http://pypi.python.org/pypi?%3Aaction=list_classifiers
-        "Development Status :: 3 - Alpha",
+        "Development Status :: 4 - Beta",
```
Comment (barrywhart) [inline setup.py:59]: 🎉🥳

**Review by barrywhart — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -14,10 +14,8 @@
 [![ReadTheDocs](https://img.shields.io/readthedocs/sqlfluff?style=flat-square&logo=Read%20the%20Docs)](https://sqlfluff.readthedocs.io)
 [![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg?style=flat-square)](https://github.com/psf/black)
 
-Bored of not having a good SQL linter that works with whichever dialect you're
-working with? SQLFluff is an extensible and modular linter designed to help you write
-good SQL and catch errors before it hits your database. SQLFluff can auto-fix most linting
-errors too.
+SQLFluff is a dialect-flexible and configurable SQL linter. Designed with ELT applications in mind, SQLFluff also works with jinja templating and dbt. SQLFluff will auto-fix most linting errors, allowing you to focus your time on what matters.
+
```
Comment (alanmcruickshank) [inline README.md:18]: Nice 👍

**Review by alanmcruickshank — APPROVED** [review]
Looks good to me! #822 is merged too now, so I'm happy for this to go when you are @NiallRees .

**Comment by NiallRees** [comment]
> Looks good to me! #822 is merged too now, so I'm happy for this to go when you are @NiallRees .

I'm going to spend a bit of time tomorrow trying to solve https://github.com/sqlfluff/sqlfluff/pull/819, but will merge by end of day whether or not I get anywhere 👍.

**Pull request merged**